proto/osmosis/lockup/lock.proto

```diff
@@ -8,74 +8,85 @@ import "cosmos/base/v1beta1/coin.proto";
+  // This value is first initialized when an unlock has started for the lock,
+  // end time being block time + duration.
   google.protobuf.Timestamp end_time = 4 [
     (gogoproto.stdtime) = true,
     (gogoproto.nullable) = false,
     (gogoproto.moretags) = "yaml:\"end_time\""
   ];
+  // Coins are the coins locked within the lock, kept in the module account.
   repeated cosmos.base.v1beta1.Coin coins = 5 [
     (gogoproto.nullable) = false,
     (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
   ];
 }
 
+// LockQueryType defines the type of the lock query that can
+// either be by duration or start time of the lock.
 enum LockQueryType {
   option (gogoproto.goproto_enum_prefix) = false;
 
-  // Queries for locks that are longer than a certain duration
   ByDuration = 0;
-  // Queries for lockups that started before a specific time
   ByTime = 1;
 }
 
+// QueryCondition is a struct used for querying locks upon different conditions.
+// Duration field and timestamp fields could be optional, depending on the
+// LockQueryType.
 message QueryCondition {
-  // type of lock query, ByLockDuration | ByLockTime
+  // LockQueryType is a type of lock query, ByLockDuration | ByLockTime
   LockQueryType lock_query_type = 1;
-  // What token denomination are we looking for lockups of
+  // Denom represents the token denomination we are looking to lock up
   string denom = 2;
-  // valid when query condition is ByDuration
+  // Duration is used to query lcoks with longer duration than the specified
+  // duration. Duration field must not be nil when the lock query type is
+  // `ByLockDuration`.
   google.protobuf.Duration duration = 3 [
     (gogoproto.stdduration) = true,
     (gogoproto.nullable) = false,
     (gogoproto.moretags) = "yaml:\"duration\""
   ];
-  // valid when query condition is ByTime
+  // Timestamp is used to locks started before the specified duration.
+  // Timestamp field must not be nil when the lock query type is `ByLockTime`.
+  // Querying locks with timestamp is currently not implemented.
   google.protobuf.Timestamp timestamp = 4 [
     (gogoproto.stdtime) = true,
     (gogoproto.nullable) = false,
     (gogoproto.moretags) = "yaml:\"timestamp\""
   ];
 }
 
-// SyntheticLock is a single unit of synthetic lockup
-// TODO: Change this to have
-// * underlying_lock_id
-// * synthetic_coin
-// * end_time
-// * duration
-// * owner
-// We then index synthetic locks by the denom, just like we do with normal
-// locks. Ideally we even get an interface, so we can re-use that same logic.
-// I currently have no idea how reward distribution is supposed to be working...
-// EVENTUALLY
-// we make a "constrained_coin" field, which is what the current "coins" field
-// is. Constrained coin field can be a #post-v7 feature, since we aren't
-// allowing partial unlocks of synthetic lockups.
+// SyntheticLock is a single unit of synthetic lockup that is created upon every
```

Comment:
Thoughts on adding what synthetic lockup means? 
This is what i got from lockup spec which could be helpful: `Synthetic lockup is creating virtual lockup where new denom is combination of original denom and synthetic suffix. At the time of synthetic lockup creation and deletion, accumulation store is also being updated and on querier side, they can query as freely as native lockup.`

Reply:
Adding a explanation here can't hurt 👍